src/scripts/templates.js

```diff
@@ -1,238 +1,192 @@
-    const localClasses = classNames(
-      globalClasses.list,
-      globalClasses.listDropdown,
-    );
 
-    return strToEl(`
-      <div
-        class="${localClasses}"
-        aria-expanded="false"
-        >
-      </div>
-    `);
-  },
-  notice(globalClasses, label, type = '') {
-    const localClasses = classNames(
-      globalClasses.item,
-      globalClasses.itemChoice,
-      {
-        [globalClasses.noResults]: type === 'no-results',
-        [globalClasses.noChoices]: type === 'no-choices',
-      },
-    );
+  choice(
+    { item, itemChoice, itemSelectable, itemDisabled, placeholder },
+    {
+      id,
+      value,
+      label,
+      groupId,
+      elementId,
+      disabled,
+      placeholder: isPlaceholder,
+    },
+    selectText,
+  ) {
+    const div = Object.assign(document.createElement('div'), {
+      id: elementId,
+      innerHTML: label,
+      className: `${item} ${itemChoice} ${
+        disabled ? itemDisabled : itemSelectable
+      } ${isPlaceholder ? placeholder : ''}`,
+    });
+    div.setAttribute('role', groupId > 0 ? 'treeitem' : 'option');
+    Object.assign(div.dataset, {
+      choice: '',
+      id,
+      value,
+      selectText,
+    });
+    if (disabled) {
+      div.dataset.choiceDisabled = '';
+      div.setAttribute('aria-disabled', 'true');
+    } else div.dataset.choiceSelectable = '';
 
-    return strToEl(`
-      <div class="${localClasses}">
-        ${label}
-      </div>
-    `);
+    return div;
+  },
+  input({ input, inputCloned }, placeholderValue) {
+    const inp = Object.assign(document.createElement('input'), {
+      type: 'text',
+      className: `${input} ${inputCloned}`,
+      autocomplete: 'off',
+      autocapitalize: 'off',
+      spellcheck: false,
+    });
+    inp.setAttribute('role', 'textbox');
+    inp.setAttribute('aria-autocomplete', 'list');
+    inp.setAttribute('aria-label', placeholderValue);
+    return inp;
+  },
+  dropdown({ list, listDropdown }) {
+    const div = document.createElement('div');
+    div.classList.add(list, listDropdown);
+    div.setAttribute('aria-expanded', 'false');
+    return div;
+  },
+  notice({ item, itemChoice, noResults, noChoices }, innerHTML, type = '') {
+    return Object.assign(document.createElement('div'), {
+      innerHTML,
+      className: `${item} ${itemChoice} ${
+        // eslint-disable-next-line no-nested-ternary
+        type === 'no-results'
+          ? noResults
+          : type === 'no-choices'
+          ? noChoices
+          : ''
+      }`,
+    });
   },
-  option(data) {
-    return strToEl(`
-      <option value="${data.value}" ${data.active ? 'selected' : ''} ${
-      data.disabled ? 'disabled' : ''
-    } ${
-      data.customProperties
-        ? `data-custom-properties=${data.customProperties}`
-        : ''
-    }>${data.label}</option>
-    `);
+  option({ label, value, customProperties, active, disabled }) {
+    const opt = new Option(label, value, false, active);
```

Comment:
Is there a reason why we don't use `createElement('option')` here? Or alternatively `new HTMLDivElement(...)` above? 🤔

Reply:
Shorter code using standard constructor that also sets attributes (selected and value) immediately

Reply:
regarding `new HTMLDivElement` - there is no such constructor.